src/amber/router/pipe/session.cr

```diff
@@ -63,8 +65,12 @@ module Amber
       end
 
       class Params < Hash(String, String)
+        property changed = false
+
         def []=(key : String | Symbol, value : V)
```

Comment:
store, an replace! aren't available to hash. `merge!` will work fine since it calls `[]=`. `merge` is fine since it doesn't actually modify the existing hash.